src/main/java/org/openjdk/jextract/clang/Index.java

```diff
@@ -61,27 +62,43 @@ public static class ParsingFailedException extends RuntimeException {
         private final ErrorCode code;
 
         public ParsingFailedException(Path srcFile, ErrorCode code) {
-            super("Failed to parse " + srcFile.toAbsolutePath().toString() + ": " + code);
-            this.srcFile = srcFile.toAbsolutePath().toString();
+            this(srcFile.toAbsolutePath().toString(), code);
+        }
+
+        public ParsingFailedException(String srcFile, ErrorCode code) {
```

Comment:
Where is this new constructor being used?

Reply:
will check and remove one of the constructors